playbooks/operationalizing/branding_console.adoc

```diff
@@ -0,0 +1,287 @@
+----
+<1> name, value are changed
+<2> name is changed
+
+[]
+* *Change copy right*
+
+.As-Is
+[source,html]
+----
+<p><strong>Welcome to Red Hat&reg; OpenShift Enterprise.</strong>
+----
+
+.To-Be
+[source,html]
+----
+<p><strong>Welcome to YOUR LOGO&reg; OpenShift Enterprise.</strong>
+----
+
+** Convert image to base64 unicode and replace default.
+(execute replace_login_logo.sh)
+
+.replace_login_logo.sh
+```
+span_badge_line=$(grep -n "span id=\"badge\"" ./new-login.html |cut -d: -f1) ;; <1>
+img_line=$(( $span_badge_line + 1 ))
+
+base64_img="<img src=\"$(echo "data:image/png;base64,$(base64 -w 0 login_logo.png)")\" alt=\"new logo\"/>" ;; <2>
+
+sed -e "$img_line,1d" -i new-login.html ;;<3>
+sed -e "$img_line i $base64_img" -i new-login.html ;;<4>
+```
+<1> Find the line of badge class.
+<2> Convert image(login_logo.png) to base64 unicode. `alt` can be modified.
+<3> Delete the default logo
+<4> Insert the new logo
+
+[]
+* *Configure master-config.yaml for new login page.*
+
+.master-config.yaml
+----
+...
+oauthConfig:
+  templates:
+    login: /etc/origin/master/new-login.html
+...
+----
+
+NOTE: The indent space is remarkably important. Sometimes, master server can not load it because of different space length.
+
+* *Now restart atomic-o penshift-master service*
+```
+# systemctl atomic-openshift-master restart
+```
+
+=== Web console logo
+After login, it is allowed to modify anything on console style using extended `css/javascrpt/images`. https://docs.openshift.com/enterprise/3.1/install_config/web_console_customization.html[The official document] explain it in detail but it could take a little bit long time to configure due to yaml syntax of master-config.xml. Hence, I recommand to follow this chapter exactly, then you can see brand new logo without trial and error.
+
+.Workflow
+. Create CSS/Image folder
+. Create css file and copy web console logo under the folders
+. Configure master-config.yaml
+
+
+
+.Steps
+Create CSS/Image folder::
+----
+# mkdir -p /etc/origin/master/stylesheet/{css,images}
+----
+
+Create css file and copy web console logo under the folders::
+[]
+.extension.css
+[source,css]
+----
+#header-logo {
+   background: url("../extensions/images/webconsole-logo.png");
+   width: 193px;
+   height: 41px;
+   margin-top: -3px;
+}
+
+#openshift-logo.navbar-brand{
+   width: 193px;
+   height: 42px;
+   padding-left: 20px;
+   padding-top: 1px;
+}
+----
+
+.Check image files
+[source,bash]
+----
+# ls -al /etc/origin/master/stylesheet/images
+webconsole-logo.png
+----
+
+Configure master-config.yaml::
+[]
+.master-config.yaml
+[source,yaml]
+----
+...
+assetConfig:
+  extensions:
+    - name: images
+      sourceDirectory:  
+          /etc/origin/master/stylesheet/images
+      extensionDevelopment: true
+      extensionStylesheets:
+    - /etc/origin/master/stylesheet/css/extension.css
+  logoutURL: ""
```

Comment:
I went through and validated setting up a custom template logo and there is minor spacing issues above. The following works:

```
assetConfig:
  extensions:
  - name: images 
    sourceDirectory: /etc/origin/master/stylesheet/images
  extensionsDevelopment: true
  extensionStylesheets:
    - /etc/origin/master/stylesheet/css/extension.css
```